### PR TITLE
add eruda support

### PR DESCRIPTION
**Pull request by adbenitez**
successor of #102

### CODE REVIEW

File: frontend/vite.config.ts
Comment (Septias) [inline frontend/vite.config.ts:30]: why not a normal else?

File: frontend/vite.config.ts
Comment (Septias) [inline frontend/vite.config.ts:34]: can't you merge this with the previous ifs?

File: frontend/vite.config.ts
Comment (adbenitez) [reply frontend/vite.config.ts:30]: because if debug == undefined it would go in that branch and return  a value instead of checking NODE_ENV in the next `if` statement

File: frontend/vite.config.ts
Comment (adbenitez) [reply frontend/vite.config.ts:34]: no, because if `debug` is set it takes priority over the value of `NODE_ENV`, but anyway we don't need that debug parameter as we want to inject Eruda dynamically not forcing it in the `vite.config.ts` file so I will just remove the debug parameter and always check `NODE_ENV`

**Comment by adbenitez** [comment]
@Septias I solved your comments, anything else preventing merging?

**Comment by Septias** [comment]
Would like to give it a look again at some point, need some time though

**Review by Simon-Laux — APPROVED** [review]
code looks good, did not test though

**Comment by Septias** [comment]
Don't merge stuff if I say I still want to object.

**Comment by adbenitez** [comment]
> Don't merge stuff if I say I still want to object.

* the changes are to a vite configuration file, not to the store source code,
* the code is small and simple, you don't need to waste hours analyzing it and trying to make it the best more beautiful and performant code ever, it was fine to merge from the beginning without any change
* I already tested it and it does what it is supposed to do, building for debug, it doesn't inject eruda, until told to do so.
* you already reviewed and I tackled the _unnecessary_ changes
* I think we better invest time in other more important stuff, if Simon already approved, and I tested and it is not the big deal of a change, it is fine to merge, you can always create an issue or a PR if you still want to keep wasting time in the vite configuration file or publish the plugin as a new project :D

**Comment by Septias** [comment]
I just wanted to test it once and give it a firm look....
Your argumentation is fine, but I think just ignoring what I said and merging is bad manner imo.

**Comment by Simon-Laux** [comment]
> Would like to give it a look again at some point, need some time though

That wording is not that clear that you want to express "please wait merging until I have found time to review it". You should be more explicit if it's important to you. ;)

**Comment by Septias** [comment]
Well if you take the comment from asiel into account, then it's pretty clear to me tbh

**Comment by Simon-Laux** [comment]
> Well if you take the comment from asiel into account, then it's pretty clear to me tbh

fair point

**Comment by adbenitez** [comment]
> but I think just ignoring what I said and merging is bad manner imo.

sorry if it felt impolite, if you said something like "I don't have time I will review it tomorrow" I would have wait for your review, but it was not clear when you would have the time, and it was not clear that it was a requirement that I needed to wait for that review when you had the time for it, 

for me what it felt bad manner is delaying the merge of my PR when as I explained above, there was not real reason to delay it so it felt a bit intentional, more if you say "yeah... maybe some day I review this when I have time and then wait for more objections, not approval", but no hard feelings

**Pull request merged**
